File: src/main/java/plus/maa/backend/service/model/CopilotSetStatus.java
package plus.maa.backend.service.model;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @author dragove
 * create on 2024-01-01
 */
@Getter
@AllArgsConstructor
public enum CopilotSetStatus {

    PUBLIC(1),
    PRIVATE(2),
    ;

    private final int code;

    /**
     * 枚举数量较少，暂时这么写
     * @param code 数据库中的数值
     * @return 枚举对象
     */
    public static CopilotSetStatus get(int code) {
        return code == 1 ? PUBLIC : PRIVATE;
    }

}

File: src/main/kotlin/plus/maa/backend/service/model/CopilotSetStatus.kt
package plus.maa.backend.service.model

/**
 * 作业/作业集的公开程度
 * 命名失误，已经改不了嘞
 *
 * @author dragove
 * create on 2024-01-01
 */
enum class CopilotSetStatus {
    /**
     * 私有，仅查看自己的作业/作业集的时候展示，其他列表页面不展示，但是通过详情接口可查询（无权限控制）
     */
    PRIVATE,

    /**
     * 公开，可以被搜索
     */
    PUBLIC,
}
